src/lib.rs

```diff
@@ -140,6 +145,32 @@ impl LndkOnionMessenger {
             return Err(());
         }
 
+        if !args.skip_version_check {
+            let version = client
+                .versioner()
+                .get_version(VersionRequest {})
+                .await
+                .expect("failed to get version")
+                .into_inner();
+    
+            if !has_version(&version, None) {
+                error!(
+                    "The LND version {} is not compatible with LNDK. Please updates to version {}.{}.{}-{}",
+                    &version.version, MIN_LND_MAJOR_VER, MIN_LND_MINOR_VER, MIN_LND_PATCH_VER, MIN_LND_PRE_RELEASE_VER
+                );
+                return Err(());
+            }
+    
+            if !has_build_tags(&version, None) {
```

Comment:
I'm thinking let's move this below the if !args.skip_version_check scope because even if skip_version_check is set, we can check if the right build tags are set

Reply:
Oh, I see. I misunderstood. I'll fix it.